src/go/k8s/internal/controller/redpanda/cluster_controller_configuration_drift.go

```diff
@@ -152,14 +154,13 @@ func (r *ClusterConfigurationDriftReconciler) Reconcile(
 		return ctrl.Result{}, fmt.Errorf("could not get cluster configuration to check drifts: %w", err)
 	}
 
-	// Since config is in sync, we assume that the current desired configuration is equal to the lastAppliedConfig and there are no invalid properties
-	patch := configuration.ThreeWayMerge(log, lastAppliedConfig, clusterConfig, lastAppliedConfig, nil, schema)
-	if patch.Empty() {
+	diff, hasDrift := hasDrift(log, lastAppliedConfig, clusterConfig, schema)
+	if !hasDrift {
 		// Nothing to do, everything in sync
 		return ctrl.Result{RequeueAfter: r.getDriftCheckPeriod()}, nil
 	}
 
-	log.Info("Detected configuration drift in the cluster")
+	log.Info("Detected configuration drift in the cluster", "patch", diff)
```

Comment:
Could this potentially log out secrets?

Reply:
since these are flagged as is_sensitive, and we remove them from the patch, i don't think so.
in addition, the entire type's .String() only outputs keys: https://github.com/redpanda-data/redpanda-operator/blob/main/src/go/k8s/pkg/resources/configuration/patch.go#L40